app/assets/stylesheets/common/components/hashtag.scss

```diff
@@ -33,16 +33,6 @@ a.hashtag {
   .hashtag-icon-placeholder {
     font-size: var(--font-down-2);
     margin: 0 0.33em 0 0.1em;
-
-    &.hashtag-missing {
```

Comment:
We still need some form of `hashtag-missing`, because if you are e.g. anon and see a hashtag that you don't have access to, you see a gap instead of a square:

![image](https://github.com/discourse/discourse/assets/920448/e1317991-6469-4ecf-bc30-11bd093f5806)

We just need to set this class if and it just needs `background-color: var(--primary-medium);` if there is a mismatch between the type + ids provided to `/hashtags/by-ids` and what is actually returned. Categories etc. that you have no permission to see will not be returned. Alternatively maybe `/hashtags/by-ids` could return some placeholder data like `{ category: [ { id: 123, access_denied: true } ] }` if you cannot access that hashtag.

![image](https://github.com/discourse/discourse/assets/920448/487a2bba-d730-4526-994e-b965b062c1bc)

Reply:
This was tricky, but I added that color to the base CSS class. What do you think?